File: src/main/java/com/fish/extendedae_plus_client/integration/recipeViewer/IRecipeViewer.java
package com.fish.extendedae_plus_client.integration.recipeViewer;

import appeng.api.stacks.GenericStack;
import com.mojang.datafixers.util.Pair;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface IRecipeViewer {
    List<GenericStack> getHoveredStacks(double mouseX, double mouseY);

    List<GenericStack> getHoveredStacks();

    List<GenericStack> getFavorites();

    @Nullable
    Pair<Boolean, Boolean> matchesKey(int mouseKey);

    default boolean isCheatMode() {
        return true;
    }

    void addFavorite(GenericStack stack);

    void setSearch(String text);
}

File: src/main/kotlin/com/fish/extendedae_plus_client/integration/recipeViewer/IRecipeViewer.kt
package com.fish.extendedae_plus_client.integration.recipeViewer

import appeng.api.stacks.GenericStack
import com.mojang.datafixers.util.Pair

interface IRecipeViewer {
    fun getHoveredStacks(mouseX: Double, mouseY: Double): MutableList<GenericStack?>

    val hoveredStacks: MutableList<GenericStack?>

    val favorites: MutableList<GenericStack?>

    fun matchesKey(mouseKey: Int): Pair<Boolean, Boolean>?

    val isCheatMode: Boolean
        get() = true

    fun addFavorite(stack: GenericStack)

    fun setSearch(text: String)
}
